From: Anshuman Khandual <anshuman.khandual@arm.com>
Subject: mm/cma: drop cma_get_name()
Date: Tue, 6 Feb 2024 09:45:18 +0530

cma_get_name() just returns cma->name without any additional
transformation unlike other helpers such as cma_get_base() and
cma_get_size().  This helper is not worth the additional indirection, and
can be dropped after replacing directly with cma->name in the sole caller
__add_cma_heap().

Link: https://lkml.kernel.org/r/20240206041518.438801-1-anshuman.khandual@arm.com
Signed-off-by: Anshuman Khandual <anshuman.khandual@arm.com>
Cc: Sumit Semwal <sumit.semwal@linaro.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 drivers/dma-buf/heaps/cma_heap.c |    2 +-
 include/linux/cma.h              |    1 -
 mm/cma.c                         |    5 -----
 3 files changed, 1 insertion(+), 7 deletions(-)

--- a/drivers/dma-buf/heaps/cma_heap.c~mm-cma-drop-cma_get_name
+++ a/drivers/dma-buf/heaps/cma_heap.c
@@ -376,7 +376,7 @@ static int __add_cma_heap(struct cma *cm
 		return -ENOMEM;
 	cma_heap->cma = cma;
 
-	exp_info.name = cma_get_name(cma);
+	exp_info.name = cma->name;
 	exp_info.ops = &cma_heap_ops;
 	exp_info.priv = cma_heap;
 
--- a/include/linux/cma.h~mm-cma-drop-cma_get_name
+++ a/include/linux/cma.h
@@ -25,7 +25,6 @@ struct cma;
 extern unsigned long totalcma_pages;
 extern phys_addr_t cma_get_base(const struct cma *cma);
 extern unsigned long cma_get_size(const struct cma *cma);
-extern const char *cma_get_name(const struct cma *cma);
 
 extern int __init cma_declare_contiguous_nid(phys_addr_t base,
 			phys_addr_t size, phys_addr_t limit,
--- a/mm/cma.c~mm-cma-drop-cma_get_name
+++ a/mm/cma.c
@@ -45,11 +45,6 @@ unsigned long cma_get_size(const struct
 	return cma->count << PAGE_SHIFT;
 }
 
-const char *cma_get_name(const struct cma *cma)
-{
-	return cma->name;
-}
-
 static unsigned long cma_bitmap_aligned_mask(const struct cma *cma,
 					     unsigned int align_order)
 {
_
